pkg/routeagent_driver/main.go

```diff
@@ -264,3 +268,96 @@ func removeInvalidSockets() {
 		}
 	}
 }
+
+// =============================================================================
+// TEMPORARY: Legacy iptables cleanup for migration to nftables
+// This entire section can be removed once migration is complete
+// =============================================================================
+
+const (
+	forwardingSubmarinerFWDChain      = "SUBMARINER-FORWARD"
+	forwardingSubmarinerMSSClampChain = "SUBMARINER-FWD-MSSCLAMP"
+)
+
+func cleanupLegacyIptables(clusterCidr []string) {
+	if !pfconfigure.IPtablesCleanupIsNeeded() {
+		return
+	}
+
+	// IP hook chains to clean up
+	ipHookChains := []iptables_cleanup.CleanupIPHookChain{
+		{
+			Table: packetfilter.TableTypeNAT,
+			Chain: &packetfilter.ChainIPHook{
+				Name:     constants.SmPostRoutingChain,
+				Type:     packetfilter.ChainTypeNAT,
+				Hook:     packetfilter.ChainHookPostrouting,
+				Priority: packetfilter.ChainPriorityFirst,
+			},
+		},
+		{
+			Table: packetfilter.TableTypeFilter,
+			Chain: &packetfilter.ChainIPHook{
+				Name:     constants.SmInputChain,
+				Type:     packetfilter.ChainTypeFilter,
+				Hook:     packetfilter.ChainHookInput,
+				Priority: packetfilter.ChainPriorityLast,
+				JumpRule: &packetfilter.Rule{
+					Proto:       packetfilter.RuleProtoUDP,
+					Action:      packetfilter.RuleActionJump,
+					TargetChain: constants.SmInputChain,
+				},
+			},
+		},
+		{
+			Table: packetfilter.TableTypeNAT,
+			Chain: &packetfilter.ChainIPHook{
+				Name:     constants.SmSelfSnatChain,
+				Type:     packetfilter.ChainTypeNAT,
+				Hook:     packetfilter.ChainHookPostrouting,
+				Priority: packetfilter.ChainPriorityMiddle,
+			},
+		},
+		{
+			Table: packetfilter.TableTypeFilter,
+			Chain: &packetfilter.ChainIPHook{
+				Name:     constants.SmForwardChain,
+				Type:     packetfilter.ChainTypeFilter,
+				Hook:     packetfilter.ChainHookForward,
+				Priority: packetfilter.ChainPriorityFirst,
+			},
+		},
+		{
+			Table: packetfilter.TableTypeFilter,
+			Chain: &packetfilter.ChainIPHook{
+				Name:     forwardingSubmarinerFWDChain,
+				Type:     packetfilter.ChainTypeFilter,
+				Hook:     packetfilter.ChainHookForward,
+				Priority: packetfilter.ChainPriorityFirst,
+			},
+		},
+		{
+			Table: packetfilter.TableTypeFilter,
+			Chain: &packetfilter.ChainIPHook{
+				Name:     forwardingSubmarinerMSSClampChain,
+				Type:     packetfilter.ChainTypeFilter,
+				Hook:     packetfilter.ChainHookForward,
+				Priority: packetfilter.ChainPriorityFirst,
+			},
+		},
+		{
+			Table: packetfilter.TableTypeRoute,
+			Chain: &packetfilter.ChainIPHook{
+				Name:     constants.SmPostRoutingMssChain,
+				Type:     packetfilter.ChainTypeRoute,
+				Hook:     packetfilter.ChainHookPostrouting,
+				Priority: packetfilter.ChainPriorityFirst,
+			},
+		},
+	}
+
+	// Execute cleanup for each IP family in the cluster
+	for _, family := range cidr.ExtractIPFamilies(clusterCidr) {
+		iptables_cleanup.CleanupChains(ipHookChains, nil, family)
+	}
```

Comment:
The MTU handler also has 2 [IP sets](https://github.com/submariner-io/submariner/blob/00184da5be3fa0df959292c5208f53bb3e3a9b3e/pkg/routeagent_driver/handlers/mtu/mtuhandler.go#L255).  Do we need to delete those as well?